### PR TITLE
Implement pretranslation request form

### LINKED ISSUE #2908: Implement pretranslation request
Spec:
https://github.com/mozilla/pontoon/blob/master/specs/0113-pretranslation-opt-in.md

### PR DESCRIPTION
**Pull request by mathjazz**
Fix #2908.

Deployed to stage for testing. Mind that clicking the big green button will actually send emails to people.

Also, before merging, should we hide the button and only start showing it once monitoring is implemented?

### CODE REVIEW

File: pontoon/projects/templates/projects/widgets/project_list.html [hunk 1]
```diff
@@ -63,8 +63,9 @@ <h4>
   </table>
   {% if request %}
     <div id="request-item-note">
-      <p>The requested projects won’t be enabled automatically: an email will be sent to Pontoon admins and team managers to evaluate the request. We’ll try to respond as soon as possible. Please make sure to get in touch with the <a href="{{ url('pontoon.teams.info', locale.code) }}">localization team</a> if you are a new contributor.</p>
+      <p class="locale-projects">The requested projects won’t be enabled automatically: an email will be sent to Pontoon admins and team managers to evaluate the request. We’ll try to respond as soon as possible. Please make sure to get in touch with the <a href="{{ url('pontoon.teams.info', locale.code) }}">localization team</a> if you are a new contributor.</p>
+      <p class="pretranslation">Pretranslation won’t be enabled automatically for the requested projects: an email will be sent to Pontoon admins and team managers to evaluate the request. We’ll try to respond as soon as possible. Note that, if a locale is not using pretranslation yet, this will require additional time to train a custom translation engine.</p>
```
Comment (mathjazz) [inline pontoon/projects/templates/projects/widgets/project_list.html:67]: I added "We’ll try to respond as soon as possible.", which we also use in the project request description.

File: pontoon/teams/templates/teams/email_request_pretranslation.jinja [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% autoescape off %}
+Please enable pretranslation for the following projects in {{ locale.name }} ({{ locale.code }}):
+{{ projects }}
+Requested by {{ user }}, {{ user_role }}:
+{{ user_url }}
+{% if locale.google_automl_model == '' %}
+IMPORTANT: Pretranslation is not enabled yet for {{ locale.name }} ({{ locale.code }}). Training a custom engine is required before enabling new projects.
```
Comment (mathjazz) [inline pontoon/teams/templates/teams/email_request_pretranslation.jinja:7]: I replaced *this locale* with `{{ locale.name }} ({{ locale.code }})`.

And moved the paragraph to the bottom. On the very top, it's more visible, but also a little out of place, because one needs to read the text that follows in order to understand what is it about.

**Comment by mathjazz** [comment]
@flodolo Could you please test the patch on stage?

**Review by flodolo — APPROVED** [review]
Seems to work fine:
* It doesn't show up on unsupported locale or for contributors.
* It's displayed for a user with translator or manager rights.

I can't test the actual email (I get an "oops" error), but I assume that's because stage is not configured to send emails?

**Comment by mathjazz** [comment]
> I can't test the actual email (I get an "oops" error), but I assume that's because stage is not configured to send emails?

Oh, right.

**Comment by mathjazz** [comment]
Flagging @eemeli for the code review.

File: pontoon/teams/static/js/request.js [hunk 3]
```diff
@@ -67,7 +72,12 @@ var Pontoon = (function (my) {
         }
 
         $('#request-item-note').toggle(show);
-        $('#request-item').toggle(show);
+
+        var title = 'Request new projects';
+        if (type === 'pretranslation') {
+          title = 'Request pretranslation';
+        }
```
Comment (eemeli) [inline pontoon/teams/static/js/request.js:79]:
```suggestion
        const title = type === 'pretranslation'
          ? 'Request pretranslation'
          : 'Request new projects';
```

File: pontoon/teams/static/js/request.js [hunk 4]
```diff
@@ -122,6 +133,29 @@ var Pontoon = (function (my) {
           },
         });
       },
+
+      requestPretranslation: function (locale, projects) {
+        $.ajax({
+          url: '/' + locale + '/request-pretranslation/',
```
Comment (eemeli) [inline pontoon/teams/static/js/request.js:139]:
```suggestion
          url: `/${locale}/request-pretranslation/`,
```

File: pontoon/teams/static/js/request.js [hunk 5]
```diff
@@ -122,6 +133,29 @@ var Pontoon = (function (my) {
...
+            csrfmiddlewaretoken: $('body').data('csrf'),
+            projects: projects,
+          },
+          success: function () {
+            Pontoon.endLoader('New pretranslation request sent.', '', 5000);
+          },
+          error: function () {
+            Pontoon.endLoader('Oops, something went wrong.', 'error');
+          },
+          complete: function () {
```
Comment (eemeli) [inline pontoon/teams/static/js/request.js:151]:
```suggestion
          success() {
            Pontoon.endLoader('New pretranslation request sent.', '', 5000);
          },
          error() {
            Pontoon.endLoader('Oops, something went wrong.', 'error');
          },
          complete() {
```

File: pontoon/settings/base.py [hunk 6]
```diff
@@ -97,6 +97,85 @@ def _default_from_email():
 # Google Cloud Translation API key
 GOOGLE_TRANSLATE_API_KEY = os.environ.get("GOOGLE_TRANSLATE_API_KEY", "")
 
+# Pontoon locale codes supported by Google Cloud AutoML Translation Project ID
+#
+# Source:
+# https://cloud.google.com/translate/automl/docs/languages#supported_codes_for_language_variants
+GOOGLE_AUTOML_SUPPORTED_LOCALES = [
```
Comment (eemeli) [inline pontoon/settings/base.py:104]: Why is this here and not in the DB?

**Review by eemeli — APPROVED** [review]
Code looks fine, some minor stuff inline.

File: pontoon/settings/base.py (hunk 6 above)
Comment (mathjazz) [reply pontoon/settings/base.py:104]: Would you rather see something like a `Locale.google_automl_enabled` boolean?

File: pontoon/teams/static/js/request.js [hunk 7]
```diff
@@ -38,26 +38,31 @@ var Pontoon = (function (my) {
...
+            .toggleClass('limited', show)
+            .toggle(show);
         } else if (type === 'team') {
           // Hide all teams and the search bar
           $('.team-list').toggle(show);
           $('.search-wrapper').toggle(show);
 
           // Show team form
           $('#request-team-form').toggle(!show);
-          Pontoon.requestItem.toggleButton(!show, 'team');
```
[the remark this answers is not in the document]
Comment (mathjazz) [reply pontoon/teams/static/js/request.js:50]: It is moved to line 57.

File: pontoon/settings/base.py (hunk 6 above)
Comment (eemeli) [reply pontoon/settings/base.py:104]: No strong opinion, but thought there could be a reason that I wasn't seeing right away. Fine either way.

**Comment by mathjazz** [comment]
> Also, before merging, should we hide the button and only start showing it once monitoring is implemented?

Added https://github.com/mozilla/pontoon/pull/2910/commits/c963b7fdadd6f781e6f9d04cd3ac1d23d8e8a561 to disable the feature.
